src/pool.rs

```diff
@@ -165,4 +172,38 @@ impl ConnectionPool {
             Err(e) => Err(Error::from(e)),
         }
     }
-}
+
+
+    fn handshake(&self, client: Client, stream: &mut PooledStream) -> Result<()> {
```

Comment:
maybe add a small doc comment here 👋

Reply:
was going to ask if it would make sense to try and DRY out this implementation of `is_master` with `client.is_master` while avoiding exposing the handshake metadata as a public option, but probably more work than it's worth.